### ISSUE TITLE
Public IP resolved as client side and not server side

### ISSUE THREAD

**Issue by zaromarco** [Lissy93/dashy#1289]
### Environment

Self-Hosted (Docker)

### System

docker on synology nas 20.10.23

### Version

latest

### Describe the problem

I added widget to get the Public IP but i found that the ip is resolved client side and not server side.
i'll try to explain:
when i'm connected from office , widget display the correct ip address , but when i'm connecting to dashy from home via vpn , widget display my home public ip address....
how to show the ip where dashy is installed? (office)
thank's

### Please tick the boxes

- [X] You have explained the issue clearly, and included all relevant info
- [X] You are using a [supported](https://github.com/Lissy93/dashy/blob/master/.github/SECURITY.md#supported-versions) version of Dashy
- [X] You've checked that this [issue hasn't already been raised](https://github.com/Lissy93/dashy/issues?q=is%3Aissue)
- [X] You've checked the [docs](https://github.com/Lissy93/dashy/tree/master/docs#readme) and [troubleshooting](https://github.com/Lissy93/dashy/blob/master/docs/troubleshooting.md#troubleshooting) guide ![important](https://img.shields.io/badge/Important!-F6094E)
- [X] You agree to the [code of conduct](https://github.com/Lissy93/dashy/blob/master/.github/CODE_OF_CONDUCT.md#contributor-covenant-code-of-conduct)

**Comment by zaromarco** [comment]
@Lissy93 do you have any news?

**Comment by Rck1984** [comment]
Running into the same issue.

I would like Dashy to show the public ip of the machine its running on, not on the client i am viewing on.

**Comment by CrazyWolf13** [comment]
I think that is expected as most of the workload happens client side, so the webpage can also run on devices like a raspberrypi.

@Lissy93 Could that be possible to get server side pub-ip?

**Comment by zaromarco** [comment]
No any news on this issue?

**Comment by Blboun3** [comment]
I guess you could try adding `useProxy: true`, thus routing the requests to the IP detection servers always via the machine on which the page is running ?

**Comment by zaromarco** [comment]
> I guess you could try adding `useProxy: true`, thus routing the requests to the IP detection servers always via the machine on which the page is running ?

thank's for the tip , tried but it doesn't work on this widget , i got an empty frame with the word flag inside and nothing else

**Comment by lissy93** [comment]
Yes, Public IP widget is pretty broken - most the upstream providers have depricated their API or changed their schema. I've got a fix coming very shortly, which fixes all this.

And, @Blboun3 is exactly correct, with `useProxy: true` it will get your server's IP. I'll add that to the docs :)

**Comment by zaromarco** [comment]
Seems it works or at least not at the loading of the dashboard because I get an empty widget and I need to click on widget reload button

**Comment by lissy93** [comment]
Try switching the provider

```yaml
- type: public-ip
  useProxy: true
  options:
    provider: ipinfo
```

If you have better results with this one, then I might change the default provider, since I've also seen something similar to what you described

Here's the list of supported providers: https://dashy.to/docs/widgets/#public-ip

**Comment by zaromarco** [comment]
> Try switching the provider
> 
> - type: public-ip
>   useProxy: true
>   options:
>     provider: ipinfo
> 
> If you have better results with this one, then I might change the default provider, since I've also seen something similar to what you described
> 
> Here's the list of supported providers: https://dashy.to/docs/widgets/#public-ip

tried with ipinfo but same problem

**Comment by lissy93** [comment]
Do any of them work for you?
They seem fine for me :/

**Comment by zaromarco** [comment]
if i look at cpu resource when i load the dashboard , after a while i see a spike and after that if look at the widget i can see the public ip
this is my docker compose file:
version: '3.8'
services:
 dashy:
    image: lissy93/dashy:latest
    container_name: Dashy
    restart: on-failure:5
    healthcheck:
      test: ['CMD', 'node', '/app/services/healthcheck']
      interval: 1m30s
      timeout: 10s
      retries: 3
      start_period: 40s
    environment:
      UID: 1026
      GID: 100
      NODE_ENV: production
    volumes:
      - /volume1/docker/dashy/my-local-conf.yml:/app/user-data/conf.yml
      - /volume1/docker/dashy/immagini:/app/user-data/item-icons/:rw
    ports:
      - 4000:8080 

let me know if it is correct or do i need to make a modification to have a better response
